DigitalLearningSolutions.Data.Tests/DataServices/ProgressDataServiceTests.cs

```diff
@@ -160,5 +160,33 @@ public void CreateNewAspProgressRecord_adds_new_record()
                 transaction.Dispose();
             }
         }
+
+        [Test]
+        public void InsertNewAspProgressForTutorialIfNoneExist_inserts_new_record()
+        {
+            // Given
+            const int tutorialId = 12732;
+            const int customisationId = 14019;
+
+            using var transaction = new TransactionScope();
+            try
+            {
+                // When
+                var initialAspProgressIds = tutorialContentTestHelper.GetAspProgressFromTutorialId(tutorialId).ToList();
+                progressDataService.InsertNewAspProgressForTutorialIfNoneExist(tutorialId, customisationId);
+                var resultAspProgressIds = tutorialContentTestHelper.GetAspProgressFromTutorialId(tutorialId).ToList();
+
+                // Then
+                using (new AssertionScope())
+                {
+                    initialAspProgressIds.Count.Should().Be(3);
+                    resultAspProgressIds.Count.Should().Be(6);

```

Comment:
I think this test needs to be a little bit more specific (or the test helper's method needs to be). Based on the counts alone, we don't know whether this could have just cloned the existing aspProgress records. I think we  need to extract the ProgressIds rather than the AspProgressIds to make sure they are unique so that we definitely are not duplicating records.

It might be nice to additionally have an entirely new test where all the progress records already have aspProgress records to test that no new records are created.